pkg/ccl/changefeedccl/sink.go

```diff
@@ -349,22 +349,47 @@ type sinkURL struct {
 	q url.Values
 }
 
+func (u *sinkURL) makeParamValues() url.Values {
+	rawValues := u.Query()
+	q := make(url.Values)
+	for k, v := range rawValues {
+		q[strings.ToLower(k)] = v
+	}
+	return q
+}
+
+// consumeParam returns the value of the parameter p from the underlying URL,
+// and deletes the parameter from the URL. The parameter name is consumed in a
+// case-insensitive manner.
 func (u *sinkURL) consumeParam(p string) string {
 	if u.q == nil {
-		u.q = u.Query()
+		u.q = u.makeParamValues()
 	}
+	p = strings.ToLower(p)
 	v := u.q.Get(p)
 	u.q.Del(p)
 	return v
 }
 
+// contains checks if the URL has the specified parameter. The parameter name is
+// checked in a case insensitive manner.
+func (u *sinkURL) contains(p string) bool {
+	if u.q == nil {
+		u.q = u.makeParamValues()
+	}
+	return u.q.Get(strings.ToLower(p)) != ""
+}
+
 func (u *sinkURL) addParam(p string, value string) {
 	if u.q == nil {
-		u.q = u.Query()
+		u.q = u.makeParamValues()
 	}
-	u.q.Add(p, value)
+	u.q.Add(strings.ToLower(p), value)
 }
 
+// consumeBool returns whether the parameter p was set in the underlying URL and
```

Comment:
[nit]: The parameter value is not called p here.

Also I think we should say `deletes the parameter from the sinkURL` to make it clear it isn't modifying the underlying URL.